pkg/provisioner/ironic/ironic.go

```diff
@@ -1219,6 +1219,35 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+
+	p.log.Info("requesting ports to delete")
+
+	allpages, err := ports.List(p.client, ports.ListOpts{NodeUUID: p.status.ID}).AllPages()
+
+	if err != nil {
+		return result, errors.Wrap(err, "failed to extract port list")
+	}
+
+	portL, err := ports.ExtractPorts(allpages)
+	for i, port := range portList {
```

Comment:
```suggestion
	for _, port := range portList {
```

go vet probably complains unless you make this `_` since the index is unused